AndrewTehan/hometask-3/rainfall_statistics.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStatistics
+  attr_reader :town, :data
+
+  def initialize(town, data)
+    @town = town
+    @data = data
+  end
+
+  def mean
+    return -1 if missing_data?
+
+    average(extract_statistic_values)
+  end
+
+  def variance
+    return -1 if missing_data?
+
+    calculate_variance(calculate_mean_differences)
+  end
```

Comment:
You could you a block there to avoid writing the same piece of code. It's not a big deal there, but ... it'll be cooler if you use some Ruby possibilities where it's suitable to do 💁🏼

```
def mean
  town_exists? { average(extract_statistic_values) }
end

def mean
  town_exists? { calculate_variance(calculate_mean_differences) }
end

def town_exists?(&calculations)
  find_town_statistics.present? ? calculation.call : return -1
end
```

Does it look better a bit? 👀